streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```
@@ -84,6 +84,27 @@ RecordQueue queue() {
         streamTime = RecordQueue.UNKNOWN;
     }
 
+    long timestamp() {
+        return streamTime;
+    }
+
+    long getPartitionTimestamp(final TopicPartition partition) {
+        if (partitionQueues.get(partition) == null) {
+            return RecordQueue.UNKNOWN;
```

I think we should throw an exception for this case, because the added partitions of the queue are fixed and should never change (and we should never request the partition time for unknown partitions -- it we do this, it would indicate a bug and would should raise it as an exception).
```
throw new NullPointerException("Partition " + partition + " not found.");
```